--- a/qfirehose.c
+++ b/qfirehose.c
@@ -412,7 +412,9 @@ int main(int argc, char **argv)
         if (optarg) {
             char filename[PATH_MAX];
-            snprintf(filename, sizeof(filename), "%.80s/qfirehose_%lu.log", optarg, time(NULL));
+            snprintf(filename, sizeof(filename),
+                     "%.80s/qfirehose_%lld.log",
+                     optarg, (long long)time(NULL));
             logfp = fopen(filename, "w");
         }
